test/e2e/e2e_test.go

```diff
@@ -0,0 +1,137 @@
+package e2e
+
+import (
+	"os"
+	"testing"
+	"time"
+
+	"github.com/xmudrii/etcdproxy-controller/pkg/apis/etcd/v1alpha1"
+	clientset "github.com/xmudrii/etcdproxy-controller/pkg/client/clientset/versioned"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+// newKubeClient creates new Kubernetes client instance based on kubeconfig file.
+func newKubeClient() (*kubernetes.Clientset, error) {
+	config, err := clientcmd.BuildConfigFromFlags("", os.Getenv("KUBECONFIG"))
+	if err != nil {
+		return nil, err
+	}
+
+	return kubernetes.NewForConfig(config)
+}
+
+// newEtcdProxyClient creates new client for etcdstorage resources.
+func newEtcdProxyClient() (*clientset.Clientset, error) {
+	config, err := clientcmd.BuildConfigFromFlags("", os.Getenv("KUBECONFIG"))
```

Comment:
I would have one helper to create the config and then do the NewForConfig call inline.  I also tolerate `NewForConfigOrDie` in tests.